modules/nlu/src/views/full/entities/SidePanelSection.tsx

```diff
@@ -62,18 +77,21 @@ export const EntitySidePanelSection: FC<Props> = props => {
         className={Classes.MINIMAL}
         icon="new-object"
         text="New entity"
-        onClick={() => setShowEntityModal(!showEntityModal)}
+        onClick={createEntity}
       />
       <SearchBar icon="filter" placeholder="filter entities" onChange={setEntitiesFilter} showButton={false} />
       <ItemList
         items={entityItems}
         onElementClicked={({ value: name }) => props.setCurrentItem({ type: 'entity', name })}
       />
-      <CreateEntityModal
+      <EntityNameModal
+        action={entityAction}
+        originalName={entityName}
+        entityIDs={props.entities.map(e => e.id)}
```

Comment:
maybe passing the whole entity would make more sens than only passing the name, this way you won't need to fetch the entity before rename and before cloning it am I right ?